crypto/bn/bn_gcd.c

```diff
@@ -612,12 +612,19 @@ int BN_gcd(BIGNUM *r, const BIGNUM *in_a, const BIGNUM *in_b, BN_CTX *ctx)
 
     /* find shared powers of two, i.e. "shifts" >= 1 */
     for (i = 0; i < r->dmax && i < g->dmax; i++) {
-        mask = ~(r->d[i] | g->d[i]);
+        mask = r->d[i] | g->d[i];
+        if ((pow2_flag == 0) | (mask == 0)) {
+            shifts += BN_BITS2 & pow2_flag;
+            continue;
```

Comment:
This makes everything non-constant time.

Reply:
Independent of input numbers (except `a==0 || b==0`, but this case checked before) it will run loop to check each bit in chunk exactly once:
```
mask    branch
0000    mask == 0, add 64 and continue
0000    mask == 0, add 64 and continue
0000    mask == 0, add 64 and continue
...
0000    mask == 0, add 64 and continue
1234    mask != 0, check each bit
5678    pow2_flag == 0, add 64&0 and continue
0000    pow2_flag == 0, add 64&0 and continue
...
```

Reply:
When we talk about constant time code, we're mostly talking about branch free code, where the for loop is the only branch code.